sqlglot/dialects/dialect.py

```diff
@@ -331,3 +332,20 @@ def create_with_partitions_sql(self, expression):
             expression.set("this", schema)
 
     return self.create_sql(expression)
+
+
+def generate_tsql_date_delta(self, e):
+    func = "DATEADD" if isinstance(e, exp.DateAdd) else "DATEDIFF"
+    return f"{func}({self.format_args(e.text('unit'), e.expression, e.this)})"
+
+
+def parse_date_delta(exp_class, unit_mapping=None):
+    def inner_func(args):
+        tsql_func = len(args) == 3
+        this = list_get(args, 2) if tsql_func else list_get(args, 0)
+        expression = list_get(args, 1) if tsql_func else list_get(args, 1)
+        unit = list_get(args, 0) if tsql_func else exp.Literal.string("DAY")
+        unit = unit_mapping.get(unit.name.lower(), unit) if unit_mapping else unit
+        return exp_class(this=this, expression=expression, unit=unit)
```

Comment:
I'm a little skeptical about this.. I think we don't want this to handle TSQL as a separate case, since no other function in the `dialects.py` file does something similar for other dialects. Maybe we should somehow make the logic / names more generic instead of the current `tsql_func` stuff?

Reply:
I see your point.
However, this parse would work as a default for some other dialects for example presto could use this function instead of the lambda it has.  All other dialects use the default parser.

Reply:
If most dialects use the default version of this, can't we simplify this by calling it inside tsql and rearranging the arguments of the produced expression or something?

I'm not sure if that's the best approach, we can wait for Toby's response on this.

Reply:
Thats how it was, but Databricks needs to do it the exact same way (and presto aswell).
So it is not only about tsql anymore.
Databricks is special since it has the option for the 2 parameter and 3 parameter function.

Reply:
yea the function shouldn't use the word tsql